src/MemoryTracing/include/MemoryTracing/MemoryInfoListener.h

```diff
@@ -13,9 +17,38 @@ namespace orbit_memory_tracing {
 class MemoryInfoListener {
  public:
   virtual ~MemoryInfoListener() = default;
-  virtual void OnSystemMemoryUsage(orbit_grpc_protos::SystemMemoryUsage memory_info) = 0;
-  virtual void OnProcessMemoryUsage(orbit_grpc_protos::ProcessMemoryUsage memory_info) = 0;
-  virtual void OnCGroupMemoryUsage(orbit_grpc_protos::CGroupMemoryUsage memory_info) = 0;
+
+  void SetSamplingStartTimestampNs(uint64_t sampling_start_timestamp_ns) {
+    sampling_start_timestamp_ns_ = sampling_start_timestamp_ns;
+  }
+  void SetSamplingPeriodNs(uint64_t sampling_period_ns) {
+    sampling_period_ns_ = sampling_period_ns;
+  }
+  void SetEnableCGroupMemory(bool enable_cgroup_memory) {
+    enable_cgroup_memory_ = enable_cgroup_memory;
+  }
+  void SetEnableProcessMemory(bool enable_process_memory) {
+    enable_process_memory_ = enable_process_memory;
+  }
+
+  void OnSystemMemoryUsage(orbit_grpc_protos::SystemMemoryUsage system_memory_usage);
+  void OnCGroupMemoryUsage(orbit_grpc_protos::CGroupMemoryUsage cgroup_memory_usage);
+  void OnProcessMemoryUsage(orbit_grpc_protos::ProcessMemoryUsage process_memory_usage);
+
+ private:
+  // This method compute the arithmetic mean of input timestamps.
+  [[nodiscard]] static uint64_t GetSynchronizedSamplingTimestamp(
+      const std::vector<uint64_t>& sampling_timestamps);
```

Comment:
Nit: This does not need to be declared in the class, does it? It could be a simple free static function in `MemoryInfoListener.cpp`, couldn't it?

Reply:
Changed accordingly! Thanks!